### PR TITLE
[JAV-286]add protect feature when call remote calls in bean initializ…

### PR DESCRIPTION
**Pull request by liubao68**
…ation

### CODE REVIEW

File: core/src/main/java/io/servicecomb/core/SystemBootListener.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+package io.servicecomb.core;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class SystemBootListener implements BootListener {
+
+  private static boolean ready = false;
```
Comment (WillemJiang) [inline core/src/main/java/io/servicecomb/core/SystemBootListener.java]: The default value of boolean is false, so you don't need assign the value.

File: core/src/test/java/io/servicecomb/core/TestSystemBootListener.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import io.servicecomb.serviceregistry.RegistryUtils;
+import io.servicecomb.serviceregistry.ServiceRegistry;
+import io.servicecomb.serviceregistry.registry.ServiceRegistryFactory;
+import mockit.Deencapsulation;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Created by l00168639 on 2017/8/10.
```
Comment (WillemJiang) [inline core/src/test/java/io/servicecomb/core/TestSystemBootListener.java]: Please update the template for it.

File: demo/demo-pojo/pojo-client/src/main/java/io/servicecomb/demo/pojo/client/BeanRpcTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package io.servicecomb.demo.pojo.client;
+
+import io.servicecomb.demo.server.Test;
+import io.servicecomb.provider.pojo.RpcReference;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by l00168639 on 2017/8/10.
```
Comment (WillemJiang) [inline demo/demo-pojo/pojo-client/src/main/java/io/servicecomb/demo/pojo/client/BeanRpcTest.java]: same issue.

File: core/src/main/java/io/servicecomb/core/SystemBootListener.java (hunk 1 above)
Comment (liubao68) [inline core/src/main/java/io/servicecomb/core/SystemBootListener.java]: I see many guidelines saying better to give a default value implicitly. e.g. some eclipse compiler and coverity tools

**Pull request merged**